lib/ecto/adapters/mysql/connection.ex

```diff
@@ -11,7 +11,13 @@ if Code.ensure_loaded?(Mariaex.Connection) do
 
     def connect(opts) do
       opts = Keyword.update(opts, :port, @default_port, &normalize_port/1)
-      Mariaex.Connection.start_link(opts)
+      case Mariaex.Connection.start_link(opts) do
+        {:ok, pid} ->
+          repo = Keyword.fetch!(opts, :repo)
+          repo.after_connect(pid)
```

Comment:
@fishcakez thanks for the review and insights. I completely agree with the shortcomings of the solution, but honestly I'm not familiar enough with Ecto/Elixir to define how to overcome them here. 

It seems from your explanation that the issue is more pervasive and need to be addressed at another level. I'm willing to work toward a better implementation if you can guide me in the right direction. Should I wrap the call in a `try/catch`, something different ?